tests/integration/snowflake/test_snowflake_adapter.py

```diff
@@ -324,3 +324,72 @@ def test_check_count_and_query(sf_adapter):
+        if SCHEMA in sf_adapter._get_all_schemas(DATABASE):
+            sf_adapter.drop_schema(database=DATABASE, name=SCHEMA)
+
+        # Test schema creation
+        sf_adapter.generate_schema(database=DATABASE, name=SCHEMA)
+        assert SCHEMA in sf_adapter._get_all_schemas(database=DATABASE)
+    finally:
+        # Clean up after test
+        sf_adapter.drop_schema(database=DATABASE, name=SCHEMA)
+
+
+def test_drop_schema(sf_adapter):
+    DATABASE, SCHEMA = "SNOWSHU_DEVELOPMENT", "DROP_SCHEMA_TEST"
+    try:
+        # Setup: Ensure the schema exists before attempting to drop it
+        if SCHEMA not in sf_adapter._get_all_schemas(DATABASE):
+            sf_adapter.generate_schema(database=DATABASE, name=SCHEMA)
+
+        # Test schema dropping
+        sf_adapter.drop_schema(database=DATABASE, name=SCHEMA)
+        assert SCHEMA not in sf_adapter._get_all_schemas(DATABASE)
+    finally:
+        # Additional cleanup, in case the test fails
+        if SCHEMA in sf_adapter._get_all_schemas(DATABASE):
+            sf_adapter.drop_schema(database=DATABASE, name=SCHEMA)
+
+
+def test_create_table(sf_adapter):
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "CREATE_TABLE_TEST", "TEST_TABLE"
+    query = "SELECT 1 AS test_col"
+    try:
+        # Setup: Create schema and ensure table does not exist
+        sf_adapter.generate_schema(name=SCHEMA, database=DATABASE)
+        if TABLE in sf_adapter._get_all_tables(DATABASE, SCHEMA):
+            sf_adapter.drop_table(name=TABLE, schema=SCHEMA, database=DATABASE)
+
+        # Test table creation
+        sf_adapter.create_table(query, name=TABLE, schema=SCHEMA, database=DATABASE)
+        assert TABLE in sf_adapter._get_all_tables(DATABASE, SCHEMA)
+    finally:
+        # Clean up after test
+        sf_adapter.drop_table(name=TABLE, schema=SCHEMA, database=DATABASE)
+        sf_adapter.drop_schema(name=SCHEMA, database=DATABASE)
+
+def test_drop_table(sf_adapter):
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "DROP_TABLE_TEST", "TEST_TABLE"
+    query = "SELECT 1 AS test_col"
+    try:
+        # Setup: Create schema and table
+        sf_adapter.generate_schema(name=SCHEMA, database=DATABASE)
+        if TABLE in sf_adapter._get_all_tables(DATABASE, SCHEMA):
+            sf_adapter.drop_table(name=TABLE, schema=SCHEMA, database=DATABASE)
+        sf_adapter.create_table(query, name=TABLE, schema=SCHEMA, database=DATABASE)
+
+        # Test table dropping
+        sf_adapter.drop_table(name=TABLE, schema=SCHEMA, database=DATABASE)
+        assert TABLE not in sf_adapter._get_all_tables(DATABASE, SCHEMA)
+    finally:
+        # Clean up: Drop table and schema if they still exist
+        if TABLE in sf_adapter._get_all_tables(DATABASE, SCHEMA):
+            sf_adapter.drop_table(name=TABLE, schema=SCHEMA, database=DATABASE)
+        if SCHEMA in sf_adapter._get_all_schemas(DATABASE):
+            sf_adapter.drop_schema(name=SCHEMA, database=DATABASE)
```

Comment:
I believe it's enough to run the drop schema statement since it has the cascade option

Reply:
That's right, unnecessary duplication here